commons/src/main/java/org/infinispan/multimap/api/MultimapCache.java

```diff
@@ -0,0 +1,91 @@
+package org.infinispan.multimap.api;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+
+/**
+ * MultimapCache provides the common building block for the two different types of multimap caches that Infinispan
+ * provides: embedded and remote.
+ * <p>
```

Comment:
We need more general documentation about this class

1. as Radim mentioned whether this supports null values or keys (it shouldn't)
2. talk about supporting duplicates and that this is determined by Object equality
3. talk about it being an async API ?
4. I would talk about all return values being views and thus updates are not reflected
5. Talk about how eviction is only counted per key and doesn't involve values

Also I forgot about this before, but we have no method to iterate over the multi map, guessing users may want this. We can always do that later though.